src/System.Configuration.ConfigurationManager/src/System/Configuration/ImplicitMachineConfigHost.cs

```diff
@@ -51,17 +51,29 @@ public override string GetStreamName(string configPath)
 
         public override Stream OpenStreamForRead(string streamName)
         {
+#if MONO 
```

Comment:
Could you upstream it without the ugly #if-defs?

Reply:
I don't think we can upstream this, the embedded machine.config only makes sense on Mono.

Reply:
The only way it could be upstreamed is if corefx allowed an embedded machine.config.  From what I understand, that is a requirement for Xamarin.

Reply:
I think we support both, the default and embedded modes this code picks just one. I think upstream would be ok to take the change if done little bit more cleaner

Reply:
> I think we support both, the default and embedded modes this code picks just one

I'm confused by this. The code in this PR supports both embedded and on-disk machine.config (`streamName` will be the full path to machine.config in this case).

Reply:
Do we really throw when we cannot find machine config?

Reply:
Yeah, the current mono System.Config does.